internal/rbd/replicationcontrollerserver_test.go

```diff
@@ -0,0 +1,155 @@
+)
+
+func TestValidateSchedulingInterval(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name     string
+		interval string
+		want     admin.Interval
+		wantErr  bool
+	}{
+		{
+			"valid interval in minutes",
+			"3m",
+			admin.Interval("3m"),
+			false,
+		},
+		{
+			"valid interval in hour",
+			"22h",
+			admin.Interval("22h"),
+			false,
+		},
+		{
+			"valid interval in days",
+			"13d",
+			admin.Interval("13d"),
+			false,
+		},
+		{
+			"invalid interval without number",
+			"d",
+			admin.Interval(""),
+			true,
+		},
+		{
+			"invalid interval without (m|h|d) suffix",
+			"12",
+			admin.Interval(""),
+			true,
+		},
+	}
+	for _, tt := range tests {
+		st := tt
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := validateSchedulingInterval(st.interval)
+			if (err != nil) != st.wantErr {
+				t.Errorf("validateSchedulingInterval() error = %v, wantErr %v", err, st.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, st.want) {
+				t.Errorf("validateSchedulingInterval() = %v, want %v", got, st.want)
+			}
+		})
+	}
+}
+
+func TestGetSchedulingDetails(t *testing.T) {
+	tests := []struct {
+		name       string
+		parameters map[string]string
+		want       admin.Interval
+		want1      admin.StartTime
+		wantErr    bool
+	}{
+		{
+			"valid parameters",
+			map[string]string{
+				imageMirroringKey:      string(imageMirrorModeSnapshot),
+				schedulingIntervalKey:  "1h",
+				schedulingStartTimeKey: "14:00:00-05:00",
+			},
+			admin.Interval("1h"),
+			admin.StartTime("14:00:00-05:00"),
+			false,
+		},
+		{
+			"valid parameters when optional startTime is missing",
+			map[string]string{
+				imageMirroringKey:     string(imageMirrorModeSnapshot),
+				schedulingIntervalKey: "1h",
+			},
+			admin.Interval("1h"),
+			admin.NoStartTime,
+			false,
+		},
+		{
+			"when mirroring mode is journal",
+			map[string]string{
+				imageMirroringKey:     "journal",
+				schedulingIntervalKey: "1h",
+			},
+			admin.NoInterval,
+			admin.NoStartTime,
+			true,
+		},
+		{
+			"when startTime is specified without interval",
+			map[string]string{
+				imageMirroringKey:      string(imageMirrorModeSnapshot),
+				schedulingStartTimeKey: "14:00:00-05:00",
+			},
+			admin.NoInterval,
+			admin.NoStartTime,
+			true,
+		},
+		{
+			"when no scheduling is specified",
+			map[string]string{
+				imageMirroringKey: string(imageMirrorModeSnapshot),
+			},
+			admin.NoInterval,
+			admin.NoStartTime,
+			false,
+		},
+	}
+	for _, tt := range tests {
+		st := tt
+		t.Run(tt.name, func(t *testing.T) {
+			got, got1, err := getSchedulingDetails(st.parameters)
+			if (err != nil) != st.wantErr {
+				t.Errorf("getSchedulingDetails() error = %v, wantErr %v", err, st.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, st.want) {
+				t.Errorf("getSchedulingDetails() got = %v, want %v", got, st.want)
+			}
+			if !reflect.DeepEqual(got1, st.want1) {
+				t.Errorf("getSchedulingDetails() got1 = %v, want %v", got1, st.want1)
```

Comment:
from output pov, got and got1may  not make sense:

I would rather do this:
```got, got1, err := getSchedulingDetails(st.parameters)
```
```
interval, starttime := getSchedulingDetails(st.parameters)
```
and in expect error:
```
interval = %v, want ..
starttime = %v, want..
```

Reply:
auto-generated test case from vscode. corrected it now.